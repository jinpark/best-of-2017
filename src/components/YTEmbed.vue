<template>
  <iframe width="100%" height="100%" :src="source" frameborder="0" gesture="media" allow="encrypted-media" allowfullscreen></iframe>
</template>

<script>

export default {
  name: 'YTEmbed',
  props: ['youtubeId'],
  computed: {
    source () {
      return `https://www.youtube.com/embed/${this.youtubeId}?rel=0&amp;showinfo=0`
    }
  }
}
</script>