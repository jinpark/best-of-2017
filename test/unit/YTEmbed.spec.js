import YTEmbed from '@/components/YTEmbed'
import { shallow } from 'vue-test-utils'
import { createRenderer } from 'vue-server-renderer'

describe('YTEmbed.vue', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallow(YTEmbed, {propsData: {youtubeId: 'abcdef'}})
  })

  it('is an iframe', () => {
    expect(wrapper.is('iframe')).toBe(true)
  })

  it('matches snapshot', () => {
    const renderer = createRenderer()
    renderer.renderToString(wrapper.vm, (err, str) => {
      if (err) throw new Error(err)
      expect(str).toMatchSnapshot()
    })
  })
})
