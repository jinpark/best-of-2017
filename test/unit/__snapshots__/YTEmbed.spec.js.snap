// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`YTEmbed.vue matches snapshot 1`] = `<iframe width="100%" height="100%" src="https://www.youtube.com/embed/abcdef?rel=0&amp;amp;showinfo=0" frameborder="0" gesture="media" allow="encrypted-media" allowfullscreen="allowfullscreen"></iframe>`;
